gourmet/Undo.py

```diff
@@ -92,7 +92,7 @@ def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=Non
             debug('Too many changes - assume 0,0',0)
             self.cindex,self.clen = 0,0
 
-    def determine_mode(self, text="", initial_text="") -> str:
+    def determine_mode(self, text: str = "", initial_text: str = "") -> str:
```

Comment:
If both arguments can be called with `None` (next two lines) it should be `Optional[str]`. Or do we actually check for empty string? What's the semantic difference between both?

Reply:
This should indeed be marked as `Optional[str]`

Reply:
This should indeed be marked as `Optional[str]`

Reply:
Done